src/arrayMethodSort.js

```diff
@@ -4,8 +4,24 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        const compareResult = compareFunction
+          ? compareFunction(this[i], this[j]) > 0
+          : String(this[i]) > String(this[j]);
+
+        if (compareResult > 0 || compareResult) {
```

Comment:
The condition `compareResult > 0 || compareResult` is redundant. Since `compareResult` is the result of a comparison, it will be a boolean value when no compareFunction is provided, and a number when it is. You can simplify this to `if (compareResult > 0)`.